lib/rust/automata/src/nfa.rs

```diff
@@ -243,21 +243,25 @@ impl Nfa {
 
     /// Convert the automata to a GraphViz Dot code for the deubgging purposes.
     pub fn as_graphviz_code(&self) -> String {
+        use std::fmt::Write;
+
         let mut out = String::new();
         for (ix, state) in self.states.iter().enumerate() {
             let opts =
                 if state.export { "" } else { "[fillcolor=\"#EEEEEE\" fontcolor=\"#888888\"]" };
-            out += &format!("node_{}[label=\"{}\"]{}\n", ix, ix, opts);
+            writeln!(out, "node_{}[label=\"{}\"]{}", ix, ix, opts).unwrap();
             for link in &state.links {
-                out += &format!(
-                    "node_{} -> node_{}[label=\"{}\"]\n",
+                writeln!(
+                    out,
+                    "node_{} -> node_{}[label=\"{}\"]",
                     ix,
                     link.target.id(),
                     link.display_symbols()
-                );
+                )
+                .unwrap();
             }
             for link in &state.epsilon_links {
-                out += &format!("node_{} -> node_{}[style=dashed]\n", ix, link.id());
+                writeln!(out, "node_{} -> node_{}[style=dashed]", ix, link.id()).unwrap();
             }
```

Comment:
I don't like these changes TBH. They are using `unwrap` now and no unwrap in the previous version. Why these changes?